src/actions/index.js

```diff
@@ -469,6 +471,19 @@ function updateOutput(
         });
     }
 
+    if(hooks.request_pre !== null) {
+        if(typeof hooks.request_pre === 'function') {
+            hooks.request_pre(JSON.stringify(payload));
```

Comment:
I think that one of the points of these hooks is the ability to modify the `payload`. So, we shouldn't make a copy of this, we should leave it up to the developer to modify it at will